File: typescript/examples/export-audit.ts
/**
 * Export a signed, offline-verifiable audit bundle to disk.
 *
 * Requires an API key with the `audit` scope.
 *
 * Run with:
 *   ATLASENT_API_KEY=ask_live_... npx tsx examples/export-audit.ts
 */

import { writeFile } from "node:fs/promises";

import { AtlaSentClient } from "@atlasent/sdk";

const apiKey = process.env.ATLASENT_API_KEY;
if (!apiKey) {
  console.error("ATLASENT_API_KEY env var is required");
  process.exit(1);
}

const client = new AtlaSentClient({ apiKey });

const bundle = await client.exportAudit({
  since: "2026-01-01T00:00:00Z",
  limit: 5000,
});

const outPath = "atlasent-audit-export.json";
await writeFile(outPath, JSON.stringify(bundle.raw, null, 2), "utf8");

const adminCount = bundle.adminLog?.length ?? 0;
console.log(
  `wrote ${bundle.evaluations.length} evaluations + ${adminCount} admin rows to ${outPath}`,
);
console.log(`signature: ${bundle.signature.slice(0, 32)}…`);

File: python/examples/export_audit.py
"""Export a signed, offline-verifiable audit bundle to disk.

Requires an API key with the ``audit`` scope.

Run with:
    ATLASENT_API_KEY=ask_live_... python examples/export_audit.py
"""

from __future__ import annotations

import json
import os
import sys

from atlasent import AtlaSentClient

api_key = os.environ.get("ATLASENT_API_KEY")
if not api_key:
    print("ATLASENT_API_KEY env var is required", file=sys.stderr)
    sys.exit(1)

with AtlaSentClient(api_key=api_key) as client:
    bundle = client.export_audit(
        since="2026-01-01T00:00:00Z",
        limit=5000,
    )

out_path = "atlasent-audit-export.json"
with open(out_path, "w", encoding="utf-8") as f:
    json.dump(bundle.model_dump(), f, indent=2)

print(
    f"wrote {len(bundle.evaluations)} evaluations "
    f"+ {0 if bundle.admin_log is None else len(bundle.admin_log)} admin rows "
    f"to {out_path}"
)
print(f"signature: {bundle.signature[:32]}…")
